src/helm/benchmark/scenarios/vision_language/mmmu_scenario.py

```diff
@@ -0,0 +1,180 @@
+        Renliang Sun and Ming Yin and Boyuan Zheng and Zhenzhu Yang and Yibo Liu and Wenhao Huang and Huan Sun
+        and Yu Su and Wenhu Chen},
+        journal={arXiv preprint arXiv:2311.16502},
+        year={2023},
+      }
+
+    Paper: https://arxiv.org/abs/2311.16502
+    Website: https://mmmu-benchmark.github.io/
+    """
+
+    MMMU_HUGGINGFACE_DATASET_NAME: str = "MMMU/MMMU"
+    MAX_NUM_IMAGES: int = 7
+
+    SUBJECTS: List[str] = [
+        "Accounting",
+        "Agriculture",
+        "Architecture_and_Engineering",
+        "Art",
+        "Art_Theory",
+        "Basic_Medical_Science",
+        "Biology",
+        "Chemistry",
+        "Clinical_Medicine",
+        "Computer_Science",
+        "Design",
+        "Diagnostics_and_Laboratory_Medicine",
+        "Economics",
+        "Electronics",
+        "Energy_and_Power",
+        "Finance",
+        "Geography",
+        "History",
+        "Literature",
+        "Manage",
+        "Marketing",
+        "Materials",
+        "Math",
+        "Mechanical_Engineering",
+        "Music",
+        "Pharmacy",
+        "Physics",
+        "Psychology",
+        "Public_Health",
+        "Sociology",
+    ]
+
+    name = "mmmu"
+    description = (
+        "Evaluate multimodal models on massive multi-discipline tasks demanding college-level "
+        "subject knowledge and deliberate reasoning ([paper](https://arxiv.org/abs/2311.16502))."
+    )
+    tags = ["vision-language"]
+
+    def __init__(self, subject: str, question_type: str):
+        super().__init__()
+        assert subject in self.SUBJECTS, f"Invalid subject: {subject}"
+        self._subject: str = subject
+
+        assert question_type in ["multiple-choice", "open"], f"Invalid question type: {question_type}"
+        self._question_type: str = question_type
+
+    def get_instances(self, output_path: str) -> List[Instance]:
+        images_path: str = os.path.join(output_path, "images", self._subject)
+        ensure_directory_exists(images_path)
+
+        instances: List[Instance] = []
+
+        # Process the validation set
+        # There seems to be a dev set, but it's unavailable through load_dataset.
+        # The test set doesn't have answers, since the MMMU competition/leaderboard uses the test set
+        for row in tqdm(load_dataset(self.MMMU_HUGGINGFACE_DATASET_NAME, split="validation", cache_dir=output_path)):
+            # Skip questions that aren't in the subject we're interested in
+            if row["question_type"] != self._question_type:
+                continue
+
+            question_id: str = row["id"]
+            if self._subject not in question_id:
+                continue
+
+            question_template: str = row["question"]
+            options: List[str] = row["options"]
+            answer: str = row["answer"]
+
+            # Create the question. Questions can have text and images interleaved
```

Comment:
Oh, I am pretty sure in the PR for Llava and OpenFlamingo we assumed the image was at the top

Reply:
I don't think we can evaluate llava and openflamingo with MMMU. There can be multiple images (up to 7) in the question and even in the answer choices.